### PR TITLE
Eloquent

### PR DESCRIPTION
**Pull request by ptrmu**
No changes are specific to eloquent so this should run on dashing also. The changes had to do with componentizing the vloc and vmap nodes.

### CODE REVIEW

File: fiducial_vlam/src/vloc_node.cpp [hunk 1]
```diff
@@ -181,26 +181,37 @@ namespace fiducial_vlam
...
+      cv::Mat mat_with_msg_data;
       if (cxt_.publish_image_marked_ &&
           count_subscribers(cxt_.image_marked_pub_topic_) > 0) {
-        color_marked = color;
+        // The toCvShare only makes ConstCvImage because they don't want
+        // to modify the original message data. I want to modify the original
+        // data so I create another CvImage that is not const and steal the
+        // image data.
+        std::shared_ptr<void const> tracked_object;
+        auto const_color_marked = cv_bridge::toCvShare(*image_msg, tracked_object);
```
Comment (clydemcqueen) [inline fiducial_vlam/src/vloc_node.cpp:201]: Interesting... I didn't know about this call.

File: fiducial_vlam/src/vloc_node.cpp [hunk 2]
```diff
@@ -181,26 +181,37 @@ namespace fiducial_vlam
...
-      cv_bridge::CvImagePtr color = cv_bridge::toCvCopy(image_msg);
+      cv_bridge::CvImagePtr gray{cv_bridge::toCvCopy(*image_msg, "mono8")};
 
       // If we are going to publish an annotated image, make a copy of
-      // the pointer to color. If no annotated image is to be published,
+      // the original message image. If no annotated image is to be published,
       // then just make an empty image pointer. The routines need to check
       // that the pointer is valid before drawing into it.
       cv_bridge::CvImagePtr color_marked;
+      cv::Mat mat_with_msg_data;
```
Comment (clydemcqueen) [inline fiducial_vlam/src/vloc_node.cpp]: Why isn't this inside the if statement?

File: fiducial_vlam/src/vloc_node.cpp [hunk 3]
```diff
@@ -141,8 +141,8 @@ namespace fiducial_vlam
 
       image_raw_sub_ = create_subscription<sensor_msgs::msg::Image>(
         cxt_.image_raw_sub_topic_,
-        rclcpp::QoS{1},  // Don't keep stale images
-        [this](const sensor_msgs::msg::Image::UniquePtr msg) -> void
+        rclcpp::ServicesQoS(),
```
Comment (clydemcqueen) [inline fiducial_vlam/src/vloc_node.cpp]: I think this sets the receive queue to 10, which can cause grief for resource-constrained control loops.

**Review by clydemcqueen — APPROVED** [review]
LGTM. 2 little things you might want to look at.

File: fiducial_vlam/src/vloc_node.cpp (hunk 2 above)
Comment (ptrmu) [reply fiducial_vlam/src/vloc_node.cpp]: Good point - Thanks you.

File: fiducial_vlam/src/vloc_node.cpp (hunk 3 above)
Comment (ptrmu) [reply fiducial_vlam/src/vloc_node.cpp]: I tried to fix this.

**Pull request merged**